File: app/src/main/java/com/mss/imagesearcher/view/imagelist/ImageListView.java
package com.mss.imagesearcher.view.imagelist;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface ImageListView extends MvpView {

    @StateStrategyType(value = AddToEndSingleStrategy.class)
    void refreshImageList();

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void openDetailActivity(int position);

    @StateStrategyType(value = AddToEndSingleStrategy.class)
    void showProgress(boolean visible);

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void showNotFoundMessage();

    @StateStrategyType(value = OneExecutionStateStrategy.class)
    void stopRefreshing();
}

File: app/src/main/java/com/mss/imagesearcher/view/imagelist/ImageListView.kt
package com.mss.imagesearcher.view.imagelist

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface ImageListView : MvpView {

    @StateStrategyType(value = AddToEndSingleStrategy::class)
    fun refreshImageList()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun openDetailActivity(position: Int)

    @StateStrategyType(value = AddToEndSingleStrategy::class)
    fun showProgress(visible: Boolean)

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showNotFoundMessage()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun stopRefreshing()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun initAdMob()

    @StateStrategyType(value = OneExecutionStateStrategy::class)
    fun showPrivacyPolicy()
}
